### DISCUSSION TITLE
Nested blocks

### DISCUSSION THREAD

**Discussion by fusionconsulting** [area17/twill#1701] [General]
I'm building a menu editor and I want to include a "heading" field type which can have child items. How can I nest blocks inside the field?

My menu module has 2 pre-defined menus (header and footer) and the form looks like this:

```
@formField('block_editor', [
    'blocks' => ['link-page', 'link-category', 'menu-header']
])
```

My blocks look like this:

```
@twillBlockTitle('Category')
@twillBlockIcon('b-accordion')

@formField('input', [
    'name' => 'title',
    'label' => 'Link title',
    'maxlength' => 60
])
@formField('browser', [
    'moduleName' => 'categories',
    'name' => 'categories',
    'label' => 'Category',
    'max' => 1,
])
```

```
@twillBlockTitle('Page')
@twillBlockIcon('website')

@formField('input', [
    'name' => 'title',
    'label' => 'Link title',
    'maxlength' => 60
])
@formField('browser', [
    'moduleName' => 'pages',
    'name' => 'pages',
    'label' => 'Page',
    'max' => 1,
])
```

These work well for what I need, the final block is the header:

```
@twillBlockTitle('Header')
@twillBlockIcon('b-link')

@formField('input', [
    'name' => 'title',
    'label' => 'Heading title',
    'maxlength' => 60
])
@formField('repeater', [
    // Not sure what this should be
])
```

How can include the category and page links to be added to the header repeater section? Or am I approaching this the wrong way?

**Comment by haringsrob** [comment]
Hey @fusionconsulting 

I think this would better fit with repeaters.

You could look here for inspiration: https://github.com/cwsdigital/twill-capsule-menus
